arch/sim/include/types.h

```diff
@@ -44,6 +44,8 @@
  * Included Files
  ****************************************************************************/
 
+#include <nuttx/compiler_stdint.h>
```

Comment:
because
* i'm not comfortable to assume that `__xxx_TYPE__` is always available.
* predefined macros are not obvious when reading the code.
* this way we can easily notice possible mistakes (like using a compiler which is configured in an incompatible way)

Reply:
So, I think that it's enough to correct typedef to ensure _xxx same as what compiler want. compiler_stdint.h is redundant and make the confusion(why define the same type twice) because the compiler can generate the mismatch warning even without including compiler_stdint.h